extension/observer/ecsobserver/exporter.go

```diff
@@ -14,10 +14,116 @@
 
 package ecsobserver
 
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultMetricsPath = "/metrics"
+)
+
 // CommonExporterConfig should be embedded into filter config.
 // They set labels like job, metrics_path etc. that can override prometheus default.
 type CommonExporterConfig struct {
 	JobName      string `mapstructure:"job_name" yaml:"job_name"`
 	MetricsPath  string `mapstructure:"metrics_path" yaml:"metrics_path"`
 	MetricsPorts []int  `mapstructure:"metrics_ports" yaml:"metrics_ports"`
 }
+
+// taskExporter converts annotated Task into PrometheusECSTarget.
+type taskExporter struct {
+	logger  *zap.Logger
+	cluster string
+}
+
+func newTaskExporter(logger *zap.Logger, cluster string) *taskExporter {
+	return &taskExporter{
+		logger:  logger,
+		cluster: cluster,
+	}
+}
+
+// ExportTasks loops a list of tasks and export prometheus scrape targets.
+// It keeps track of error but does NOT stop when error occurs.
+// The returned targets are all valid and the error(s) are mainly for generating metrics.
```

Comment:
Not sure what it means for targets to be valid and there to be errors

Reply:
Also if it's only for metrics maybe the metrics can be generated here instead of relying on `multierr`

Reply:
Added a util func to print the error as structured log, which is useful if your log backend supporting filtering based on log fields. 

I checked [obsreport](https://github.com/open-telemetry/opentelemetry-collector/tree/main/obsreport) package for generating component metrics, it seems it has no out of box support for extension metrics? It's more about traces for other component types (receiver, processor and exporter).